src/Components/Components/src/Rendering/HtmlRenderer.cs

```diff
@@ -47,20 +47,20 @@ protected override Task UpdateDisplayAsync(in RenderBatch renderBatch)
         /// <param name="componentType">The type of the <see cref="IComponent"/>.</param>
         /// <param name="initialParameters">A <see cref="ParameterCollection"/> with the initial parameters to render the component.</param>
         /// <returns>A <see cref="Task"/> that on completion returns a sequence of <see cref="string"/> fragments that represent the HTML text of the component.</returns>
-        public async Task<IEnumerable<string>> RenderComponentAsync(Type componentType, ParameterCollection initialParameters)
+        public async Task<RenderedHtmlResult> RenderComponentAsync(Type componentType, ParameterCollection initialParameters)
```

Comment:
Suggested name `ComponentRenderedText`

Reply:
We've tried to avoid `Html` appearing and names, and IMO `Result` could be replaced with something much more descriptive or just removed.